File: go/internal/scoring/testdata/structural-v1/typescript/distinct_two.ts
export function distinctTwo(value: number): number {
  if (value > 0) {
    if (value % 2 === 0) return value * 2;
  }
  return value;
}

export function anotherDifficult(value: number): number {
  if (value < 0) {
    if (value % 2 !== 0) return -value;
  }
  return value;
}

File: go/internal/scoring/testdata/structural-v1/rust/distinct_two.rs
pub fn distinct_two(value: i32) -> i32 {
    if value > 0 {
        if value % 2 == 0 { return value * 2; }
    }
    value
}

pub fn another_difficult(value: i32) -> i32 {
    if value < 0 {
        if value % 2 != 0 { return -value; }
    }
    value
}
